waiter/src/waiter/auth/authentication.clj

```diff
@@ -44,14 +49,16 @@
 
   (create-auth-handler [_ request-handler]
     (fn anonymous-handler [request]
-      (let [request' (assoc request
-                       :authorization/user run-as-user
-                       :authenticated-principal run-as-user)]
-        (request-handler request')))))
+      (-> request
+          (assoc :authorization/user run-as-user
+                 :authenticated-principal run-as-user)
+          request-handler
+          (add-cached-auth password run-as-user)
+          cookie-support/cookies-async-response))))
```

Comment:
Can we extract this `(-> ...)` block intoa helper function in `authentication.clj`. Spnego has the same block. We can then also make `add-cached-auth` private.